social_media/accounts/views.py

```diff
@@ -0,0 +1,117 @@
+from django.contrib.auth import authenticate
+from django.shortcuts import render
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.authtoken.models import Token
+from .serializer import AccountRegistrationSerializer,AccountSerializer
+from .models import User
+
+
+# Create your views here.
+class UserDetailsView(APIView):
+
+    def get(self,request,username):
+        try:
+            user = User.objects.get(username=username)
+        except User.DoesNotExist:
+            newdict = {'not_found': True}
+            return Response(data={'error': {'message': 'user not found'}}, status=status.HTTP_404_NOT_FOUND)
+
+        account_serializer = AccountSerializer(user)
+        return Response(account_serializer.data)
+
+
+
+
+
+class Login(APIView):
+    def post(self, request):
+        data = request.data
+        try:
+            username = data["email"]
+            password = data["password"]
+        except:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        # check if a user with such username or email exist
+        try:
```

Comment:
Nope, same comments as above. Don't use try-catch, instead query the db using .filter() or .exists()